test/test_bad_files.py

```diff
@@ -0,0 +1,164 @@
+    cache = default_shared_state.cache
+    config.ALLOW_BAD_RULES.set("1")
+    config.ALLOW_BAD_REFERENCES.set("1")
+    crds.getreferences(HST_HEADER, observatory='hst', context='hst_0282.pmap', reftypes=['pfltfile'])
+    _, err = capsys.readouterr()
+    out_to_check = """Recommended reference 'l2d0959cj_pfl.fits' of type 'pfltfile' is designated scientifically \
+invalid."""
+    default_shared_state.cleanup()
+    assert out_to_check in err
+
+
+
+@mark.bad_files
+def test_bad_references_fast_mode(default_shared_state):
+    cache = default_shared_state.cache
+    config.ALLOW_BAD_RULES.reset()
+    config.ALLOW_BAD_REFERENCES.reset()
+    val = crds.getreferences(HST_HEADER, observatory='hst', context='hst_0282.pmap', reftypes=['pfltfile'], fast=True)
+    val_to_check = {}
+    val_to_check['pfltfile'] = f'{cache}/references/hst/l2d0959cj_pfl.fits'
+    default_shared_state.cleanup()
+    assert val == val_to_check
+
+
+
+@mark.bad_files
+def test_bad_references_bestrefs_script_error(caplog, default_shared_state, hst_data):
+    config.ALLOW_BAD_RULES.reset()
+    config.ALLOW_BAD_REFERENCES.reset()
+    args = f"crds.bestrefs --new-context hst_0282.pmap --files {hst_data}/j8btxxx_raw_bad.fits"
+    with caplog.at_level(logging.INFO, logger="CRDS"):
+        BestrefsScript(args)()
+        out = caplog.text
+        print('out is')
+        print(out)
+    default_shared_state.cleanup()
+    out_to_check = f"""No comparison context or source comparison requested.
+No file header updates requested;  dry run.  Use --update-bestrefs to update FITS headers.
+===> Processing /home/runner/work/crds/crds/test/data/hst/j8btxxx_raw_bad.fits
+Failed processing '/home/runner/work/crds/crds/test/data/hst/j8btxxx_raw_bad.fits' : Failed computing bestrefs \
+for data '/home/runner/work/crds/crds/test/data/hst/j8btxxx_raw_bad.fits' with respect to 'hst_0282.pmap' : \
+Recommended reference 'l2d0959cj_pfl.fits' of type 'pfltfile' is designated scientifically invalid.
+
+1 errors
+0 warnings
+3 infos""".splitlines()
+    for line in out_to_check:
+        assert line in out
+
+
+@mark.bad_files
+def test_bad_references_bestrefs_script_warning(caplog, default_shared_state, hst_data):
+    config.ALLOW_BAD_RULES.set("1")
+    config.ALLOW_BAD_REFERENCES.set("1")
+    args = f"crds.bestrefs --new-context hst_0282.pmap --files {hst_data}/j8btxxx_raw_bad.fits"
+    with caplog.at_level(logging.INFO, logger="CRDS"):
+        BestrefsScript(args)()
+        out = caplog.text
+        print('out is')
+        print(out)
+    default_shared_state.cleanup()
+    out_to_check = f"""No comparison context or source comparison requested.
+No file header updates requested;  dry run.  Use --update-bestrefs to update FITS headers.
+ ===> Processing /home/runner/work/crds/crds/test/data/hst/j8btxxx_raw_bad.fits
+Failed processing '/home/runner/work/crds/crds/test/data/hst/j8btxxx_raw_bad.fits' : Failed computing bestrefs for \
```

Comment:
Ended up having to change this path to '/home/runner/work/crds' to work for runtests but in my container (and pycharm) I get 'home/developer'. Is there something I can modify to get what would be expected here when I run my local tests?

Reply:
Note that the actual check, below is `<something> in <something>`, which means that you can modify the relevant lines in the expected output to have only a subset of that line.

I see in the original doctest-based that there were ways of flagging ignore areas. This level of complexity is not necessary.